src/Microsoft.AspNetCore.Identity/IdentityOptions.cs

```diff
@@ -11,6 +13,18 @@ namespace Microsoft.AspNetCore.Builder
     /// </summary>
     public class IdentityOptions
     {
+        public IdentityOptions(IEnumerable<IConfigureOptions<IdentityOptions>> configureOptions = null)
```

Comment:
Trust me I don't like this at all...  But its a 'taste' thing...  There will be a more sane looking options pattern where you call

`services.AddOptions<YourOption>()` which registers a singleton and does the mechanical gunk for you...  

But, hopefully everyone has the same reaction as you, and we will internally use the sugar :)